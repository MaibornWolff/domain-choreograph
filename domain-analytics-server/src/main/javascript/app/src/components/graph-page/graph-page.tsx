import * as React from 'react';
import { MapInteractionCSS } from 'react-map-interaction';
import { GraphSelection } from '~components/graph-selection/graph-selection';
import { GraphContainer } from '~containers/graph.container';
import { NavigationBarContainer } from '~containers/navigation-bar.container';
import { SideBarContainer } from '~containers/side-bar.container';
import { ChoreoGraph } from '~types/choreo-graph';
import { styled } from '~utils/styled';

const Wrapper = styled('div')`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  flex: 1 1 0;
`;

const GraphWrapper = styled('div')`
  flex: 1 1 0;
  display: flex;
  align-items: center;
  flex-direction: column;
`;

const PaddingWrapper = styled('div')`
  padding: 1rem;
`;

const ContentWrapper = styled('div')`
  position: relative;
  display: flex;
  height: 100%;
`;

export interface GraphPageProps {
  graph: ChoreoGraph | null;
  executionContext: string;
  scope?: string;
}

export interface GraphPageState {
}

export class GraphPage extends React.Component<GraphPageProps, GraphPageState> {
  public render() {
    const { scope, graph, executionContext: executionContextKey } = this.props;
    if (graph == null) {
      return <p>Graph is required</p>;
    }
    const executionContext = graph.executionContexts[executionContextKey];
    if (!executionContext) {
      return <p>Execution Context with key {executionContextKey} not found</p>;
    }
    const scopes = executionContext.scopes;

    let children: JSX.Element;
    if (scopes.length === 0) {
      children = <p>No Scopes</p>;
    } else if (scope) {
      children = scopes.includes(scope)
        ? (
          <MapInteractionCSS key={scope}>
            <PaddingWrapper>
              <GraphContainer scope={scope}/>
            </PaddingWrapper>
          </MapInteractionCSS>
        )
        : <p>Scope {scope} not found</p>;
    } else {
      children = scopes.length === 1
        ? (
          <MapInteractionCSS key={scopes[0]}>
            <PaddingWrapper>
              <GraphContainer scope={scopes[0]}/>
            </PaddingWrapper>
          </MapInteractionCSS>
        )
        : <GraphSelection executionContext={executionContextKey} scopes={scopes}/>;
    }

    return (
      <Wrapper>
        <NavigationBarContainer/>
        <ContentWrapper>
          <GraphWrapper>
            {children}
          </GraphWrapper>
          <SideBarContainer/>
        </ContentWrapper>
      </Wrapper>
    );
  }
}
