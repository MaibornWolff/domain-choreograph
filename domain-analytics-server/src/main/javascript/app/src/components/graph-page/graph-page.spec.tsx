import { shallow } from 'enzyme';
import * as React from 'react';
import { GraphPage } from './graph-page';

describe('GraphPage', () => {
  it('should render without graph', () => {
    expect(shallow(<GraphPage graph={null} executionContext={'Application'}/>)).toMatchSnapshot();
  });

  it('should render with graph', () => {
    const graph = {
      nodes: {
        'A': {
          id: 'A',
          name: 'A',
          scope: 'root',
          value: {},
        },
        'F': {
          id: 'F',
          name: 'F',
          scope: 'root',
          value: {},
        },
        '0-B': {
          id: '0-B',
          name: '0-B',
          scope: '0',
          value: {},
        },
        '0-C': {
          id: '0-C',
          name: '0-C',
          scope: '0',
          value: {},
        },
        '0-D': {
          id: '0-D',
          name: '0-D',
          scope: '0',
          value: {},
        },
        '0-E': {
          id: '0-E',
          name: '0-E',
          scope: '0',
          value: {},
        },
        '1-B': {
          id: '1-B',
          name: '1-B',
          scope: '1',
          value: {},
        },
        '1-C': {
          id: '1-C',
          name: '1-C',
          scope: '1',
          value: {},
        },
        '1-D': {
          id: '1-D',
          name: '1-D',
          scope: '1',
          value: {},
        },
        '1-E': {
          id: '1-E',
          name: '1-E',
          scope: '1',
          value: {},
        }
      },
      dependencies: [
        {
          src: 'A',
          target: 'F'
        },
        {
          src: '0-B',
          target: '0-C'
        },
        {
          src: '0-B',
          target: '0-D'
        },
        {
          src: '0-C',
          target: '0-E'
        },
        {
          src: '0-D',
          target: '0-E'
        },
        {
          src: '1-B',
          target: '1-C'
        },
        {
          src: '1-B',
          target: '1-D'
        },
        {
          src: '1-C',
          target: '1-E'
        },
        {
          src: '1-D',
          target: '1-E'
        },
        {
          src: 'A',
          target: '0-D'
        },
        {
          src: 'A',
          target: '1-D'
        }
      ],
      scopes: {
        root: {
          id: 'root',
          executionContext: 'Application',
          nodes: [
            'A',
            'F'
          ]
        },
        0: {
          id: '0',
          executionContext: 'F',
          nodes: [
            '0-B',
            '0-C',
            '0-D',
            '0-E'
          ]
        },
        1: {
          id: '1',
          executionContext: 'F',
          nodes: [
            '1-B',
            '1-C',
            '1-D',
            '1-E'
          ]
        }
      },
      executionContexts: {
        Application: {
          id: 'Application',
          scopes: [
            'root'
          ]
        },
        F: {
          id: 'F',
          scopes: [
            '0',
            '1'
          ]
        }
      }
    };
    expect(shallow(<GraphPage graph={graph} executionContext={'Application'}/>)).toMatchSnapshot();
  });

  it('should render with scope', () => {
    const graph = {
      nodes: {
        'A': {
          id: 'A',
          name: 'A',
          scope: 'root',
          value: {},
        },
        'F': {
          id: 'F',
          name: 'F',
          scope: 'root',
          value: {},
        },
        '0-B': {
          id: '0-B',
          name: '0-B',
          scope: '0',
          value: {},
        },
        '0-C': {
          id: '0-C',
          name: '0-C',
          scope: '0',
          value: {},
        },
        '0-D': {
          id: '0-D',
          name: '0-D',
          scope: '0',
          value: {},
        },
        '0-E': {
          id: '0-E',
          name: '0-E',
          scope: '0',
          value: {},
        },
        '1-B': {
          id: '1-B',
          name: '1-B',
          scope: '1',
          value: {},
        },
        '1-C': {
          id: '1-C',
          name: '1-C',
          scope: '1',
          value: {},
        },
        '1-D': {
          id: '1-D',
          name: '1-D',
          scope: '1',
          value: {},
        },
        '1-E': {
          id: '1-E',
          name: '1-E',
          scope: '1',
          value: {},
        }
      },
      dependencies: [
        {
          src: 'A',
          target: 'F'
        },
        {
          src: '0-B',
          target: '0-C'
        },
        {
          src: '0-B',
          target: '0-D'
        },
        {
          src: '0-C',
          target: '0-E'
        },
        {
          src: '0-D',
          target: '0-E'
        },
        {
          src: '1-B',
          target: '1-C'
        },
        {
          src: '1-B',
          target: '1-D'
        },
        {
          src: '1-C',
          target: '1-E'
        },
        {
          src: '1-D',
          target: '1-E'
        },
        {
          src: 'A',
          target: '0-D'
        },
        {
          src: 'A',
          target: '1-D'
        }
      ],
      scopes: {
        root: {
          id: 'root',
          executionContext: 'Application',
          nodes: [
            'A',
            'F'
          ]
        },
        0: {
          id: '0',
          executionContext: 'F',
          nodes: [
            '0-B',
            '0-C',
            '0-D',
            '0-E'
          ]
        },
        1: {
          id: '1',
          executionContext: 'F',
          nodes: [
            '1-B',
            '1-C',
            '1-D',
            '1-E'
          ]
        }
      },
      executionContexts: {
        Application: {
          id: 'Application',
          scopes: [
            'root'
          ]
        },
        F: {
          id: 'F',
          scopes: [
            '0',
            '1'
          ]
        }
      }
    };
    expect(shallow(<GraphPage graph={graph} executionContext={'Application'} scope={'root'}/>)).toMatchSnapshot();
  });

  it('should render with graph without scopes', () => {
    const graph = {
      nodes: {
        A: {
          id: 'A',
          name: 'A',
          scope: 'root',
          value: {},
        },
        F: {
          id: 'F',
          name: 'F',
          scope: 'root',
          value: {},
        },
      },
      dependencies: [
        {
          src: 'A',
          target: 'F'
        },
      ],
      scopes: {
      },
      executionContexts: {
        Application: {
          id: 'Application',
          scopes: [ ]
        },
      }
    };
    expect(shallow(<GraphPage graph={graph} executionContext={'Application'}/>)).toMatchSnapshot();
  });

});
