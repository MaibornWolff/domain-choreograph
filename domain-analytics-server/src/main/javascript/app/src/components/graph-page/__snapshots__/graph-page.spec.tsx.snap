// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphPage should render with graph 1`] = `
<Wrapper>
  <withRouter(renderNavigationBarWithRoutes) />
  <ContentWrapper>
    <GraphWrapper>
      <MapInteractionCSS
        key="root"
      >
        <PaddingWrapper>
          <Connect(Graph)
            scope="root"
          />
        </PaddingWrapper>
      </MapInteractionCSS>
    </GraphWrapper>
    <Connect(WithTheme(SideBar)) />
  </ContentWrapper>
</Wrapper>
`;

exports[`GraphPage should render with graph without scopes 1`] = `
<Wrapper>
  <withRouter(renderNavigationBarWithRoutes) />
  <ContentWrapper>
    <GraphWrapper>
      <p>
        No Scopes
      </p>
    </GraphWrapper>
    <Connect(WithTheme(SideBar)) />
  </ContentWrapper>
</Wrapper>
`;

exports[`GraphPage should render with scope 1`] = `
<Wrapper>
  <withRouter(renderNavigationBarWithRoutes) />
  <ContentWrapper>
    <GraphWrapper>
      <MapInteractionCSS
        key="root"
      >
        <PaddingWrapper>
          <Connect(Graph)
            scope="root"
          />
        </PaddingWrapper>
      </MapInteractionCSS>
    </GraphWrapper>
    <Connect(WithTheme(SideBar)) />
  </ContentWrapper>
</Wrapper>
`;

exports[`GraphPage should render without graph 1`] = `
<p>
  Graph is required
</p>
`;
